### PR TITLE
Initial refactoring of shared code for KeyVault

### PR DESCRIPTION
**Pull request by heaths**
Previously, shared projects didn't work cross-platform but it seems like the *.shproj is only used in VS's solution explorer. The actual file sharing is done via `<Import>` and should work cross-platform, so long as the pipelines don't use .sln files with _dotnet_ (ones I sampled don't).

### CODE REVIEW

File: sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs [hunk 1]
```diff
@@ -51,14 +51,17 @@ public SecretClient(Uri vaultUri, TokenCredential credential)
         /// <param name="vaultUri">Endpoint URL for the Azure Key Vault service.</param>
         /// <param name="credential">Represents a credential capable of providing an OAuth token.</param>
         /// <param name="options">Options that allow to configure the management of the request sent to Key Vault.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="vaultUri"/> or <paramref name="credential"/> is null.</exception>
         public SecretClient(Uri vaultUri, TokenCredential credential, SecretClientOptions options)
         {
-            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(credential));
+            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(vaultUri));
             options = options ?? new SecretClientOptions();
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Secrets/src/SecretClient.cs]: Good catch.  We should also check credential here though, and remove the check from the internal class.

File: sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs [hunk 2]
```diff
@@ -47,14 +49,17 @@ public KeyClient(Uri vaultUri, TokenCredential credential)
         /// <param name="vaultUri">Endpoint URL for the Azure Key Vault service.</param>
         /// <param name="credential">Represents a credential capable of providing an OAuth token.</param>
         /// <param name="options">Options that allow to configure the management of the request sent to Key Vault.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="vaultUri"/> or <paramref name="credential"/> is null.</exception>
         public KeyClient(Uri vaultUri, TokenCredential credential, KeyClientOptions options)
         {
-            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(credential));
+            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(vaultUri));
             options = options ?? new KeyClientOptions();
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Keys/src/KeyClient.cs]: Same comment as in the SecretClient ctor

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ public CertificateClient(Uri vaultUri, TokenCredential credential)
 
         public CertificateClient(Uri vaultUri, TokenCredential credential, CertificateClientOptions options)
         {
-            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(credential));
+            _vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(vaultUri));
```
Comment (schaabs) [inline sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs]: >_vaultUri = vaultUri ?? throw new ArgumentNullException(nameof(vaultUri)); [](start = 12, length = 74)

You can revert this change I have fixed this in the CertificateClient implementation which I should PR this afternoon

**Review by schaabs — APPROVED** [review]
:shipit:

File: sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs (hunk 3 above)
Comment (heaths) [reply sdk/keyvault/Azure.Security.KeyVault.Certificates/src/CertificateClient.cs]: If it'll help avoid merge conflicts, I certainly will.
